src/libraries/Common/src/System/Security/Cryptography/PasswordBasedEncryption.cs

```diff
@@ -1113,5 +1109,62 @@ private static RC2 CreateRC2()
 
             return RC2.Create();
         }
+
+        private static byte[] DeriveKey(
+            byte[] password,
+            ReadOnlySpan<byte> salt,
+            int iterationCount,
+            HashAlgorithmName prf,
+            int outputLength)
+        {
+#if NETCOREAPP
+            return Rfc2898DeriveBytes.Pbkdf2(password, salt, iterationCount, prf, outputLength);
+#else
+            using (Rfc2898DeriveBytes pbkdf2 = OpenPbkdf2(password, salt.ToArray(), iterationCount, prf))
+            {
+                return pbkdf2.GetBytes(outputLength);
+            }
+#endif
+        }
+
+        private static Rfc2898DeriveBytes OpenPbkdf2(
+            byte[] password,
+            byte[] salt,
+            int iterationCount,
+            HashAlgorithmName prf)
+        {
+#if NETCOREAPP || NETSTANDARD2_1_OR_GREATER || NET472_OR_GREATER
+#pragma warning disable CA5379
+            return new Rfc2898DeriveBytes(
+                password,
+                salt,
+                iterationCount,
+                prf);
+#pragma warning restore CA5379
+#else
+            if (prf == HashAlgorithmName.SHA1)
+            {
+#pragma warning disable CA5379
+                return new Rfc2898DeriveBytes(password, salt, iterationCount);
+#pragma warning restore CA5379
+            }
+
+            try
+            {
+                return (Rfc2898DeriveBytes)Activator.CreateInstance(
```

Comment:
I think it would make sense to have a managed version of PBKDF2 that can work with any hash algorithm. Or is this just a placeholder?

Reply:
Rfc2898DeriveBytes _is_ a managed implementation that works with any hash algorithm.  But the necessary ctor is only defined for .NET Core, .NET 5+, .NET Standard 2.1, and .NET Framework 4.7.2+.  So, all the places that can call it at compile time do so, above.

We get here when compiled for .NET Standard 2.0 or .NET Framework prior to 4.7.2 (both of which happen in Microsoft.Bcl.Cryptography).  If the running target is .NET Core / .NET 5+ or a high enough .NET Framework (which should be true everywhere) then Activator.CreateInstance will find the matching ctor at runtime and call it.  So, it only fails when the actual runtime is .NET Framework 4.6.2, 4.7(.0), or 4.7.1 (and the PRF isn't SHA-1).  If that becomes a problem then we can duplicate the code.

Reply:
I know. Sorry if my question was unclear: I was asking if your intention was to eventually include a private managed version for M.B.Cryptography or continue to throw. Sounds like "throw". I do think it would be worth eventually including a managed implementation to handle SHA256, since Windows started surfacing that in their export dialog.